src/foam/nanos/ruler/CompositeRuleAction.js

```diff
@@ -0,0 +1,48 @@
+/**
```

Comment:
This is the correct license header, since you don't work for Google.

/**
 * @license
 * Copyright 2019 The FOAM Authors. All Rights Reserved.
 * http://www.apache.org/licenses/LICENSE-2.0
 */